src/utils/schemaPrinter.ts

```diff
@@ -89,15 +103,61 @@ type Options = {
   sortInterfaces?: boolean;
   sortUnions?: boolean;
   sortEnums?: boolean;
+  sortCustom?: SortTC;
+  sortCustomOptions?: SortTCOptions;
 };
 
 export type SchemaPrinterOptions = Options;
 
-export type SchemaComposerPrinterOptions = Options & {
-  include?: string[];
-  exclude?: string[];
-  omitDirectiveDefinitions?: boolean;
-};
+export type SchemaComposerPrinterOptions = Options & SchemaFilterTypes;
+
+interface PrinterFilterOptions {
+  optPrinter: SchemaPrinterOptions,
+  optFilter: SchemaFilterTypes,
+}
+
+export function printSortAlpha(
+  tc1: NamedTypeComposer<any>,
+  tc2: NamedTypeComposer<any>,
+) {
+  return tc1.getTypeName().localeCompare(tc2.getTypeName())
+}
+
+function sortGetPositionOfType(
+  tc: NamedTypeComposer<any>,
+  rootTypes: string[] = [],
+): number {
+  switch (true) {
+    case tc instanceof ScalarTypeComposer: return 1;
+    case tc instanceof EnumTypeComposer: return 3;
+    case tc instanceof InterfaceTypeComposer: return 4;
+    case tc instanceof InputTypeComposer: return 5;
+    case tc instanceof ObjectTypeComposer:
+      return rootTypes.includes(tc.getTypeName()) ? 2 : 6;
```

Comment:
We also can improve sorting for ROOT types - print Query, Mutation, Subscription in the strict order:

```ts
if (tc instanceof ObjectTypeComposer) {
  const typename =  tc.getTypeName();
  if (rootTypes[0] === typename) return 2.0;
  if (rootTypes[1] === typename) return 2.1;
  if (rootTypes[2] === typename) return 2.2;

  return 6;
}
```